L1Trigger/TrackFindingTMTT/src/Histos.cc

```diff
@@ -0,0 +1,3970 @@
+          if (tpRecoedMap[&tp]) {  // This truth particle was successfully fitted.
+            hisFitTPinvptForEff_[fitName]->Fill(1. / tp.pt());
+            hisFitTPptForEff_[fitName]->Fill(tp.pt());
+            hisFitTPetaForEff_[fitName]->Fill(tp.eta());
+            hisFitTPphiForEff_[fitName]->Fill(tp.phi0());
+            // Plot also production point of all good reconstructed TP.
+            hisFitTPd0ForEff_[fitName]->Fill(fabs(tp.d0()));
+            hisFitTPz0ForEff_[fitName]->Fill(fabs(tp.z0()));
+            // Also plot efficiency to perfectly reconstruct the track (no fake hits)
+            if (tpPerfRecoedMap[&tp]) {  // This truth particle was successfully fitted with no incorrect hits.
+              hisPerfFitTPinvptForEff_[fitName]->Fill(1. / tp.pt());
+              hisPerfFitTPptForEff_[fitName]->Fill(tp.pt());
+              hisPerfFitTPetaForEff_[fitName]->Fill(tp.eta());
+            }
+            if (tp.useForAlgEff()) {  // Check TP is good for algorithmic efficiency measurement.
+              hisFitTPinvptForAlgEff_[fitName]->Fill(1. / tp.pt());
+              hisFitTPptForAlgEff_[fitName]->Fill(tp.pt());
+              hisFitTPetaForAlgEff_[fitName]->Fill(tp.eta());
+              hisFitTPphiForAlgEff_[fitName]->Fill(tp.phi0());
+              // Plot also production point of all good reconstructed TP.
+              hisFitTPd0ForAlgEff_[fitName]->Fill(fabs(tp.d0()));
+              hisFitTPz0ForAlgEff_[fitName]->Fill(fabs(tp.z0()));
+              // Plot sector number to understand if looser cuts are needed in certain regions.
+              hisFitTPphisecForAlgEff_[fitName]->Fill(iPhiSec_TP);
+              hisFitTPetasecForAlgEff_[fitName]->Fill(iEtaReg_TP);
+              // Also plot efficiency to perfectly reconstruct the track (no fake hits)
+              if (tpPerfRecoedMap[&tp]) {
+                hisPerfFitTPinvptForAlgEff_[fitName]->Fill(1. / tp.pt());
+                hisPerfFitTPptForAlgEff_[fitName]->Fill(tp.pt());
+                hisPerfFitTPetaForAlgEff_[fitName]->Fill(tp.eta());
+                hisPerfFitTPphisecForAlgEff_[fitName]->Fill(iPhiSec_TP);
+                hisPerfFitTPetasecForAlgEff_[fitName]->Fill(iEtaReg_TP);
+                // Efficiency inside jets.
+                if (tp.tpInJet()) {
+                  hisPerfFitTPinvptForAlgEff_inJetPtG30_[fitName]->Fill(1. / tp.pt());
+                }
+                if (tp.tpInHighPtJet()) {
+                  hisPerfFitTPinvptForAlgEff_inJetPtG100_[fitName]->Fill(1. / tp.pt());
+                }
+                if (tp.tpInVeryHighPtJet()) {
+                  hisPerfFitTPinvptForAlgEff_inJetPtG200_[fitName]->Fill(1. / tp.pt());
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  //=== Produce plots of tracking efficiency after HT or after r-z track filter (run at end of job).
+
+  TFileDirectory Histos::plotTrackEfficiency(string tName) {
```

Comment:
Done. (Though the CPU performance of the Histos class is not critical, since histogramming is disabled via python cfg during MC production. The histograms are only enabled by the L1 track group for algo development).